src/verify.rs

```diff
@@ -72,26 +73,62 @@ fn compile_and_test(exercise: &Exercise, run_mode: RunMode) -> Result<bool, ()>
     let result = compilation.run();
     progress_bar.finish_and_clear();
 
-    match result {
-        Ok(_) => {
-            if let RunMode::Interactive = run_mode {
-                Ok(prompt_for_completion(&exercise))
-            } else {
-                Ok(true)
-            }
-        }
+    let output = match result {
+        Ok(output) => output,
         Err(output) => {
             warn!(
                 "Testing of {} failed! Please try again. Here's the output:",
                 exercise
             );
             println!("{}", output.stdout);
+            return Err(());
+        }
+    };
+
+    let comparison_result = match exercise.expected_output() {
+        None => Ok(()),
+        Some(expected) => {
+            if expected == output.stdout {
+                Ok(())
+            } else {
+                Err((expected, &output.stdout))
+            }
+        }
+    };
+
+    match comparison_result {
+        Ok(_) => {
+            success!("Successfully tested {}!", exercise);
```

Comment:
Re-added this line, a regression introduced in https://github.com/rust-lang/rustlings/pull/271 :see_no_evil:

Reply:
Still re-added in the last version.